### PR TITLE
Add github action for App Distribution

### PR DESCRIPTION
**Pull request by jeremydurham**
Add Github action for App Distribution

### CODE REVIEW

File: .github/workflows/appdistribution.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    needs: pod-lib-lint
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup Bundler
+      run: scripts/setup_bundler.sh
+    - name: PodLibLint App Distribution Cron
+      run: |
+       scripts/third_party/travis/retry.sh scripts/pod_lib_lint.rb FirebaseAppDistribution.podspec \
+         --platforms=${{ matrix.target }} ${{ matrix.flags }} \
+         --ignore-local-podspecs=FirebaseAppDistribution.podspec
```
Comment (paulb777) [inline .github/workflows/appdistribution.yml:52]: The `--ignore-local-podspecs` should not be here.

File: .github/workflows/appdistribution.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      matrix:
+        target: [ios]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup Bundler
+      run: scripts/setup_bundler.sh
+    - name: Build and test
+      run: |
+       scripts/third_party/travis/retry.sh scripts/pod_lib_lint.rb FirebaseAppDistribution.podspec \
+         --platforms=${{ matrix.target }} --ignore-local-podspecs=FirebaseAppDistribution.podspec
```
Comment (paulb777) [inline .github/workflows/appdistribution.yml:30]: Remove `--ignore-local-podspecs=FirebaseAppDistribution.podspec`

File: .github/workflows/appdistribution.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  appdistribution-cron-only:
+    if: github.event_name == 'schedule' && github.repository != 'FirebasePrivate/firebase-ios-sdk'
+
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        target: [ios, tvos, macos]
```
Comment (paulb777) [inline .github/workflows/appdistribution.yml:38]: Do you support tvos and macos? If so, they should be in the non-schedule job also.

Comment (jeremydurham) [reply .github/workflows/appdistribution.yml:38]: No, only iOS for now. I'll remove these.

File: .github/workflows/appdistribution.yml (hunk 2 above)
Comment (jeremydurham) [reply .github/workflows/appdistribution.yml]: Done!

File: .github/workflows/appdistribution.yml (hunk 1 above)
Comment (jeremydurham) [reply .github/workflows/appdistribution.yml]: Done!

**Comment by jeremydurham** [comment]
@paulb777 I got the green light on this (pod lint ran and passed), and when adding the badge I got a failure because of gha issues. I am currently merging this from branch -> branch, as opposed to mainline. Are you OK with us merging this into our branch (app-distribution-eap-reloaded)?

**Comment by paulb777** [comment]
@jeremydurham I'm fine with whatever merge policies work best for branches.  GHA green is encouraged but not required. And it's especially optional when GHA is as bad as it has been this week.

File: README.md [hunk 4]
```diff
@@ -3,6 +3,7 @@
 [![Platform](https://img.shields.io/cocoapods/p/Firebase.svg?style=flat)](https://cocoapods.org/pods/Firebase)
 
 [![Actions Status][gh-abtesting-badge]][gh-actions]
+[![Actions Status][gh-appdistribution-badge]][gh-actions]
```
Comment (paulb777) [inline README.md:6]: There should be another change at the bottom of the file.

Comment (jeremydurham) [reply README.md:6]: 🤦 - done!

**Pull request merged**